pico/configdriven/services/src/main/java/io/helidon/pico/configdriven/services/DefaultConfigBeanRegistryProvider.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2023 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.pico.configdriven.services;
+
+import io.helidon.builder.config.spi.BasicConfigBeanRegistry;
+import io.helidon.builder.config.spi.ConfigBeanRegistryProvider;
+import io.helidon.common.LazyValue;
+import io.helidon.common.Weight;
+import io.helidon.common.Weighted;
+
+/**
+ * Service-loaded provider for {@link ConfigBeanRegistry}.
+ */
+@Weight(Weighted.DEFAULT_WEIGHT)
+public class DefaultConfigBeanRegistryProvider implements ConfigBeanRegistryProvider {
+    static final LazyValue<ConfigBeanRegistry> INSTANCE = LazyValue.create(DefaultConfigBeanRegistry::new);
+
+    /**
+     * Service loader based constructor.
+     *
+     * @deprecated this is a Java ServiceLoader implementation and the constructor should not be used directly
+     */
+    @Deprecated
+    public DefaultConfigBeanRegistryProvider() {
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public <T extends BasicConfigBeanRegistry> T configBeanRegistry() {
```

Comment:
The signature is wrong. The service provider interface must return `ConfigBeanRegistry`, this can return `BasicConfigBeanRegistry`, but it cannot return this signature, as you may just cast is to a descendant and you would get a class cast exception (even though the code passes compilation). 
This signature makes this an untyped signature